py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,427 @@
+                                 " coming up!", 'warning', self.warn_timeout)
+
+    def _format_timedelta(self, index, time_to):
+        """
+        Formats the dict time_to containg days/hours/minutes until an event starts
+        into a string according to time_to_formatted.
+
+        Returns: The formatted string.
+        """
+        time_to_formatted = ''
+
+        if time_to['total_minutes'] <= self.time_to_max:
+            time_to_formatted = self.py3.safe_format(
+                self.format_time_to,
+                {'days': time_to['days'],
+                 'hours': time_to['hours'],
+                 'minutes': time_to['minutes']})
+
+        return time_to_formatted
+
+    def _check_button_open(self, index, url):
+        """
+        Checks if the button associated with opening an event's URL in a browser has
+        been pressed. If so, that URL is opened in the default web browser.
+        """
+        if self.button == self.button_open and self.button_index == index:
+            self.py3.command_run('xdg-open ' + url)
+
+            self.no_update = False
+            self.button = None
+
+    def _check_button_toggle(self, index):
+        """
+        Checks if the button associated with toggling event format between
+        format_event and format_event_full has been pressed. If so,
+        we toggle the format.
+        """
+        if self.button == self.button_toggle and self.button_states[index]:
+            curr_event_format = self.format_event_full
+        else:
+            curr_event_format = self.format_event
+
+        self.no_update = False
+
+        return curr_event_format
+
+    def _build_response(self):
+        """
+        Builds the composite reponse to be output by the module by looping through all
+        events and formatting the necessary strings.
+
+        Returns: A composite containing the individual response for each event.
+        """
+        responses = []
+        index = 0
+
+        for event in self.events:
+            summary = event.get('summary')
+            location = event.get('location')
+            description = event.get('description')
+            url = event['htmlLink']
+
+            start_dt = gstr_to_datetime(event['start'].get('dateTime'))
+            end_dt = gstr_to_datetime(event['end'].get('dateTime'))
+
+            start_time_str = datetime_to_str(start_dt, self.format_time)
+            end_time_str = datetime_to_str(end_dt, self.format_time)
+
+            start_date_str = datetime_to_str(start_dt, self.format_date)
+            end_date_str = datetime_to_str(end_dt, self.format_date)
+
+            time_to = delta_time(start_dt)
+
+            self._check_warn_threshold(time_to, summary)
+
+            if time_to['days'] < 0:
+                time_to = delta_time(end_dt)
+                time_to_formatted = '{} {}'.format(self._format_timedelta(index, time_to),
+                                                   self.event_time_suffix)
```

Comment:
I'm not sure if this is the best way as this is likely hardcoded and the option `event_time_suffix` is simple. Users can't customize much here... For one, we couldn't add icon or anything before the time... eg, no `Time left: 50 minutes` And there is a hard space between them too.

Reply:
Ok, so this was also causing the weird output earlier. You're right, hardcoding it sucks. What I did now is better I think. There's format_time_to and format_time_left now. The former is used to format time_delta when the event isn't in progress, and the latter when the event is in progress. That way the user can indicate however they like the difference between the two. Good?